test/Core.Test/Services/CollectionServiceTests.cs

```diff
@@ -57,26 +67,36 @@ await sutProvider.GetDependency<IEventService>().Received()
         organization.UseGroups = true;
         sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(organization.Id).Returns(organization);
         var utcNow = DateTime.UtcNow;
+        var userList = users.ToList();
+        var groupList = groups.ToList();
+        userList.First().Manage = true;
+
 
-        await sutProvider.Sut.SaveAsync(collection, groups, users);
+        await sutProvider.Sut.SaveAsync(collection, groupList, userList);
 
-        await sutProvider.GetDependency<ICollectionRepository>().Received().CreateAsync(collection, groups, users);
+        await sutProvider.GetDependency<ICollectionRepository>().Received()
+            .CreateAsync(collection, Arg.Any<List<CollectionAccessSelection>>(),
+                Arg.Is<List<CollectionAccessSelection>>(list => list.Any(item => item.Manage == true)));
         await sutProvider.GetDependency<IEventService>().Received()
             .LogCollectionEventAsync(collection, EventType.Collection_Created);
         Assert.True(collection.CreationDate - utcNow < TimeSpan.FromSeconds(1));
         Assert.True(collection.RevisionDate - utcNow < TimeSpan.FromSeconds(1));
     }
 
     [Theory, BitAutoData]
-    public async Task SaveAsync_NonDefaultId_ReplacesCollectionInRepository(Collection collection, Organization organization, SutProvider<CollectionService> sutProvider)
+    public async Task SaveAsync_NonDefaultId_ReplacesCollectionInRepository(Collection collection, Organization organization, CollectionAccessSelection userAccess, SutProvider<CollectionService> sutProvider)
     {
         var creationDate = collection.CreationDate;
         sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(organization.Id).Returns(organization);
         var utcNow = DateTime.UtcNow;
+        userAccess.Manage = true;
+        List<CollectionAccessSelection> userList = new() { userAccess };
 
-        await sutProvider.Sut.SaveAsync(collection);
+        await sutProvider.Sut.SaveAsync(collection, null, userList);
 
-        await sutProvider.GetDependency<ICollectionRepository>().Received().ReplaceAsync(collection, null, null);
+        await sutProvider.GetDependency<ICollectionRepository>().Received().ReplaceAsync(collection,
+            Arg.Any<List<CollectionAccessSelection>>(),
```

Comment:
question: Should this group list arg still be `null`?

Reply:
I'm honestly not 100% sure: i also expected that passing in `null` would work but NSubstitute was tripping on the params with an error message similar to this:

`NSubstitute.Exceptions.AmbiguousArgumentsException
Cannot determine argument specifications to use. Please use specifications for all arguments of the same type.
Method signature:
    CreateAsync(Collection, IEnumerable<CollectionAccessSelection>, IEnumerable<CollectionAccessSelection>)
Method arguments (possible arg matchers are indicated with '*'):
    CreateAsync(Collection, *<null>*, *<null>*)`

Reply:
Ah, well that is an annoying error 😦 

I think we still want to have the explicit check for `null` instead of `Any`, so we should be able to swap that with something like the following:

```Arg.Is<IEnumerable<CollectionAccessSelection>>(l => l == null)```

Reply:
Updated!